syft/generic/pointers/pointer_plan.py

```diff
@@ -43,18 +47,53 @@ def __init__(
             description=description,
         )
 
+    # Make PointerPlan compatible with multi pointers
+    @property
+    def location(self):
+        n_locations = len(self._locations)
+        if n_locations != 1:
+            return self._locations
+        else:
+            return self._locations[0]
+
+    @location.setter
+    def location(self, new_location: Union[AbstractWorker, List[AbstractWorker]]):
+        if isinstance(new_location, AbstractWorker):
+            self._locations = [new_location]
+        else:
+            self._locations = new_location
+
+    @property
+    def id_at_location(self):
+        n_ids = len(self._ids_at_location)
+        if n_ids != 1:
+            return self._ids_at_location
+        else:
+            return self._ids_at_location[0]
+
+    @id_at_location.setter
+    def id_at_location(self, new_id_at_location):
+        if isinstance(new_id_at_location, (int, str)):
+            self._ids_at_location = [new_id_at_location]
+        else:
+            self._ids_at_location = new_id_at_location
+
     def __call__(self, *args, **kwargs):
         """
         Transform the call on the pointer in a request to evaluate the
         remote plan
         """
-        result_ids = [sy.ID_PROVIDER.pop()]
+        if len(self._locations) == 1:
+            location = self.location
+        else:
+            location = args[0].location
 
-        response = self.request_execute_plan(self.location, result_ids, *args)
+        result_ids = [sy.ID_PROVIDER.pop()]
```

Comment:
since this is plural should pop happen more than once?

Reply:
Good point, theoretically this is here to allow multiple ids in the result ids but in 95% of the case you only have one. We have a fallback mechanism for simple operations to populate this list when there are more than 1 object to return, but currently this hasn't been studied for plans. By default we always use only one, because we build pointers for each id and we don't want stale pointer